shared/util.ts

```diff
@@ -1,14 +1,14 @@
 import { GameVariable, IUpgrade, IUpgradeEffect, VariableModFunction } from "./types";
 import { allUpgrades } from "./upgrades";
 
-function pluck<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
-  const arrCop: Array<T> = arr;
-  arr = [];
+export function pluckOne<T>(arr: Array<T>, pred: (thing: T) => Boolean): T | null {
+  const arrCop: Array<T> = [...arr];
+  arr.length = 0;
```

Comment:
this should actually modify the array properly

Reply:
Any idea why `arr = []` didn't work? I'm not super well-versed in the peculiarities of Javascript but I'm really surprised that didn't reset the array but `arr.length = 0` does.

Reply:
I actually have no idea. I'm guessing `arr = []` actually resets the pointer so it doesn't reference the original array anywhere, whereas `array.length = 0` actually references the original array and mutates it. As with many things in JavaScript, for me it comes down to 🤷‍♂️

Reply:
Javascript feels like the language of bizarre undefined behavior